File: ios/SweetSheetModule.swift
import ExpoModulesCore
import SwiftUI

public class SweetSheetModule: Module {
  public func definition() -> ModuleDefinition {
    Name("SweetSheet")
    View(SweetSheetView.self) {
      Events("onDismiss")
      Prop("isPresented") { (view: SweetSheetView, isPresented: Bool?) in
        view.props.isPresented = isPresented ?? false
      }
      Prop("detents") {(view: SweetSheetView, detents: [Any]?) in
        view.props.detents = detents ?? []
      }
      Prop("hideDragIndicator") {(view: SweetSheetView, hideDragIndicator: Bool?) in
        view.props.hideDragIndicator = hideDragIndicator ?? false
      }
      Prop("cornerRadius") {(view: SweetSheetView, cornerRadius: Double?) in
        view.props.cornerRadius = cornerRadius.map { CGFloat($0) }
      }
    }
  }
}

File: android/src/main/java/expo/modules/sweetsheet/SweetSheetModule.kt
package expo.modules.sweetsheet

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class SweetSheetModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("SweetSheet")

    View(SweetSheetView::class) {
      Events("onDismiss")
      Prop("isPresented") { view: SweetSheetView, isPresented: Boolean? ->
        view.updateIsPresented(isPresented ?: false)
      }
      Prop("hideDragIndicator") { view: SweetSheetView, hideDragIndicator: Boolean? ->
        view.updateHideDragIndicator(hideDragIndicator ?: false)
      }
      Prop("cornerRadius") { view: SweetSheetView, cornerRadius: Int? ->
        view.updateCornerRadius(cornerRadius ?: null)
      }
    }
  }
}
